### PR TITLE
MP-8897 - Validate that Owned Entities cannot be auditable

### LINKED ISSUE #628: [NoxSolution] Validate that Owned Entities cannot be auditable
remove  the code that checks that the entity isOwned to not generate Auditable fields, for example:

src/Nox.Generator/Application/Dto/EntityDto.template.cs
src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs

### LINKED ISSUE #665: Resolve Crypto Cash API compilation issues when setting IsAudited flag in Employee entity

### PR DESCRIPTION
**Pull request by dejobratic**

### CODE REVIEW

File: src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs [hunk 1]
```diff
@@ -13,6 +16,17 @@ public EntityPersistenceValidator(string entityName)
...
                 .NotEmpty()
                 .WithMessage(ep => string.Format(ValidationResources.EntityPersistenceTableNameEmpty, entityName));
+
+            RuleFor(ep => ep.IsAudited)
+                .NotEqual(true)
+                .When(ep => IsOwnedByAnyOtherEntity(entityName, entities))
+                .WithMessage(ep => string.Format(ValidationResources.EntityOwnedCannotBeAuditable, entityName));
         }
+
+        private static bool IsOwnedByAnyOtherEntity(string ownedEntityName, IEnumerable<Entity>? parentEntities)
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: Let's rename the "ownedEntityName" variable to "entityName" as it is accepting any Entity Name and then it checks if it is owned or not.

Comment (gjorgji-grgovski-authoritypartners) [reply src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: Let's rename the "parentEntities" variable to "entities" these are all entities defined for this domain.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

File: src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs (hunk 1 above)
Comment (dejobratic) [reply src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: Done.

File: src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs [hunk 2]
```diff
@@ -13,6 +16,17 @@ public EntityPersistenceValidator(string entityName)
             RuleFor(ep => ep.TableName)
                 .NotEmpty()
                 .WithMessage(ep => string.Format(ValidationResources.EntityPersistenceTableNameEmpty, entityName));
+
+            RuleFor(ep => ep.IsAudited)
+                .NotEqual(true)
+                .When(ep => IsOwnedByAnyOtherEntity(entityName, entities))
```
Comment (rochar) [inline src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: the entity has a IsOwnedEntity property, that you can use without querying the list of entitites

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, just a small improvement proposal

File: src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs (hunk 2 above)
Comment (rochar) [reply src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: please update from main, thks

Comment (dejobratic) [reply src/Nox.Solution/Validation/Domain/EntityPersistenceValidator.cs]: Updated. Thanks for the info.

**Review by rochar — APPROVED** [review]

**Pull request merged**